extensions/amp-consent/0.1/test/test-consent-state-manager.js

```diff
@@ -477,13 +503,19 @@ describes.realWin('ConsentStateManager', {amp: 1}, (env) => {
         expect(requestBody.consentStateValue).to.equal('accepted');
         expect(requestBody.consentString).to.equal('old');
         expect(requestBody.consentMetadata).to.be.undefined;
-        yield instance.update(CONSENT_ITEM_STATE.ACCEPTED, 'new', {});
+        yield instance.update(
```

Comment:
I'm fine with this since you've already added the metadata to the test. 
However I also find it unnecessary with this unit test and many tests above. The test here is to specifically cover consentString change, in the spirit of unit test `consentMetadata` value should not affect the test. 
Ideally the optional value `consentMetadata` will only be added to related tests to keep the unit tests clean. 
To add test coverage: It's great to add more unit tests target `consentMetadata` functionalities. And add the value to the integration test case.

Reply:
Good point. I will be more selective in the places where I test future metadata functionality. 

> It's great to add more unit tests target consentMetadata functionalities

Are there particular functionalities I am missing? I think most of the functionality is handled by test-consent-info.js

Added integration test.

Reply:
The test coverage on functionalities looks good to me. 

Please remind me, I think we didn't check local storage value for a reason.
We can add e2e test coverage once we expose API to get metadata. (or even use analytics variable substitution)

Reply:
> Please remind me, I think we didn't check local storage value for a reason.

We felt that e2e test was not the place to access localstorage, and instead relied upon outcome of test (blocked elements vs non-blocked) as the indicator of localstorage values. We also used analytics substitutions to verify as that was an end behavior of the test. 

We have now have the exposed API and once we finish all the metadata value implementations, I will create e2e cases to test (#28634)